### PR TITLE
 Include limited metadata in 'DirEntry' objects

### PR DESCRIPTION
**Pull request by piscisaureus**
For consistency, this PR also modifies std's `WalkEntry` interface to extend `DirEntry` with an additional `path` field.

### CODE REVIEW

File: std/fs/walk.ts [hunk 1]
```diff
@@ -16,25 +16,25 @@ export interface WalkOptions {
...
-  if (skip && skip.some((pattern): boolean => !!filename.match(pattern))) {
+  if (skip && skip.some((pattern): boolean => !!name.match(pattern))) {
     return false;
   }
   return true;
 }
 
 export interface WalkEntry {
-  filename: string;
+  name: string;
```
Comment (nayeemrmn) [inline std/fs/walk.ts:37]: Maybe it should be `path`?

File: cli/js/lib.deno.ns.d.ts [hunk 2]
```diff
@@ -1388,8 +1388,9 @@ declare namespace Deno {
    * Requires `allow-read` permission. */
   export function realpath(path: string): Promise<string>;
 
-  export interface DirEntry extends FileInfo {
+  export interface DirEntry {
     name: string;
+    info: FileInfo;
```
Comment (bartlomieju) [inline cli/js/lib.deno.ns.d.ts:1393]: If I remember correctly we want to make `stat` optional - so this field would be optional as well; is that correct @piscisaureus? If so we should do that before 1.0 as making it `info?` is breaking change

File: std/fs/expand_glob.ts [hunk 3]
```diff
@@ -73,8 +73,8 @@ export async function* expandGlob(
   const excludePatterns = exclude
     .map(resolveFromRoot)
     .map((s: string): RegExp => globToRegExp(s, globOptions));
-  const shouldInclude = (filename: string): boolean =>
-    !excludePatterns.some((p: RegExp): boolean => !!filename.match(p));
+  const shouldInclude = (name: string): boolean =>
```
Comment (bartlomieju) [inline std/fs/expand_glob.ts:76]: Not sure why we're changing `filename` to `name`...

**Comment by ry** [comment]
I wonder if we can restrict `readdir` further and have it return a `Iterable<string>`... we could just leave the `stat` operation to users.

**Pull request merged**